core/src/impl/Kokkos_HostSpace.cpp

```diff
@@ -91,6 +91,9 @@ void *HostSpace::impl_allocate(
     size_t size = (arg_alloc_size + alignment - 1) / alignment * alignment;
 #ifdef KOKKOS_COMPILER_MSVC
     ptr = _aligned_malloc(size, alignment);
+#elif defined(__APPLE__)
+    // FIXME_APPLE MacOS X doesn't provide std::aligned_alloc before 10.15
+    posix_memalign(&ptr, alignment, size);
```

Comment:
`posix_memalign` is undefined unless `alignment >= sizeof(void*)`. Is that a precondition of this function already, or do you need to increase the alignment if it's small?

Reply:
The alignment is always 64 as far as I can tell https://github.com/kokkos/kokkos/blob/ca3687e637c12225e824ee626e69fc9fea8cebd2/core/src/Kokkos_Macros.hpp#L190-L192

Looks like `alignment`
> must be a power of two and a multiple `of sizeof(void *)`

So I suppose we're good.

Reply:
Unless it's user-provided as compile definition (which probably almost never happens). I added a `static_assert` so we are safe either way.